### PR TITLE
Improve the coverage recipe

### PR DESCRIPTION
**Pull request by jamestalmage**
Easier to review here:

https://github.com/jamestalmage/ava/blob/fix-coverage-recipe/docs/recipes/code-coverage.md

### CODE REVIEW

File: docs/recipes/code-coverage.md [hunk 1]
```diff
@@ -41,24 +60,67 @@ First, we'll need a babel configuration. This will vary from developer to develo
 }
 ```
 
-Note that in development mode, we need to specify a sourcemap when we transpile our code, and in production this is unnecessary. So for your production script, use an environment other than development; for example:
+There are two important things to note from the example above.
+
+  1. We ignore test files because AVA already handles transpiling tests for you.
+
+  2. We specify `inline` source maps for development. This is important for properly generating coverage. Using the `env` section of the Babel configuration allows us to disable source maps for production builds
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: Should end in a `.`

File: docs/recipes/code-coverage.md [hunk 2]
```diff
@@ -41,24 +60,67 @@ First, we'll need a babel configuration. This will vary from developer to develo
...
+    "build": "BABEL_ENV=production babel --out-dir=dist index.js"
+  }
 }
 ```
 
-> WARNING: Windows will choke when you set environment variables with BABEL_ENV=production like that. We recommend using [`cross-env`] to eliminate this problem under Windows. Note, the use of [`cross-env`] in the other OS runs as well. If you want your project to work cross-platform, use it.
+> WARNING: `BABEL_ENV=production` does not work on Windows, you must use the `set` keyword  (`set BABEL_ENV=production`).  For cross platform builds, check out [`cross-env`].
 
-To cover ES6, simply prepend your test script with `nyc` and the `--babel` flag. This npm script will then handle our code coverage and testing:
+Note that the build script really has very little to do with AVA, and is just a demonstration of how to use Babels `env` configuration to manipulate your config so it is compatible with AVA.
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: Babels => Babel's

File: docs/recipes/code-coverage.md [hunk 3]
```diff
@@ -41,24 +60,67 @@ First, we'll need a babel configuration. This will vary from developer to develo
...
+
+### Use the Babel require hook
+
+To use the babel require hook, add `babel-core/register` to the `require` section of you AVA config in `package.json`.
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: babel => Babel

File: docs/recipes/code-coverage.md [hunk 4]
```diff
@@ -89,22 +153,25 @@ This will output a HTML file to the `coverage` directory.
...
+First, you must login to [coveralls.io] and activate your repository.
+ 
+Once that is done, add [`coveralls`] as a development dependency:
 
-```
-npm install coveralls --save-dev
-```
+  ```
+  $ npm install coveralls --save-dev
+  ```
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: Don't indent

File: docs/recipes/code-coverage.md [hunk 5]
```diff
@@ -89,22 +153,25 @@ This will output a HTML file to the `coverage` directory.
...
 Then add the following to your `.travis.yml`:
 
-```
-after_success:
-	- './node_modules/.bin/nyc report --reporter=text-lcov | ./node_modules/.bin/coveralls'
-```
+  ```yaml
+  after_success:
+    - './node_modules/.bin/nyc report --reporter=text-lcov | ./node_modules/.bin/coveralls'
+  ```
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: Don't indent

File: docs/recipes/code-coverage.md [hunk 6]
```diff
@@ -1,17 +1,28 @@
 # Code coverage
 
-As AVA [spawns the test files][isolated-env], you can't use [`istanbul`] for code coverage; instead, you can achieve this with [`nyc`] which is basically [`istanbul`] with sub-process support. So, firstly we'll need to install it:
+As AVA [spawns the test files][isolated-env], you can't use [`istanbul`] for code coverage; instead, you can achieve this with [`nyc`] which is basically [`istanbul`] with sub-process support. 
+
+## Setup
+
+First install nyc:
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: nyc => NYC to be consistent with the usage further down.

File: docs/recipes/code-coverage.md [hunk 7]
```diff
@@ -89,22 +153,25 @@ This will output a HTML file to the `coverage` directory.
...
+  ```yaml
+  after_success:
+    - './node_modules/.bin/nyc report --reporter=text-lcov | ./node_modules/.bin/coveralls'
+  ```
 
-Your coverage report will then appear on coveralls shortly after the CI service completes.
+Your coverage report will then appear on coveralls shortly after Travis completes.
 
 [`babel`]:      https://github.com/babel/babel
+[coveralls.io]: https://coveralls.io/
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: `https://coveralls.io/` => `https://coveralls.io`

**Comment by sindresorhus** [comment]
Generally looks good. Added some minor nitpick.

**Comment by vdemedes** [comment]
LGTM

**Pull request merged**